domain/services/model.go

```diff
@@ -142,18 +142,20 @@ func (s *ModelFactory) BlockDevice() *blockdeviceservice.WatchableService {
 }
 
 // Application returns the model's application service.
-func (s *ModelFactory) Application(secrets applicationservice.SecretService) *applicationservice.WatchableService {
+func (s *ModelFactory) Application() *applicationservice.WatchableService {
 	return applicationservice.NewWatchableService(
 		applicationstate.NewApplicationState(changestream.NewTxnRunnerFactory(s.modelDB),
 			s.logger.Child("application"),
 		),
+		secretstate.NewState(changestream.NewTxnRunnerFactory(s.modelDB),
+			s.logger.Child("application"),
+		),
```

Comment:
I love this set of changes in this PR, but I do wonder if we should pass in the whole kit and kaboodle. I can see it both ways, the interface prevents additional access to the implementation, but we have to instantiate the whole secret state.

I'm happy with the progress, we can always have a `secretstate.NewDeleteState` at a later date.

I am apprehensive about pushing DeleteSecrets into the application state as it is already getting large (it's getting bigger as we speak), but we may be faced with that when we remove RunAtomic and we want to keep transactionality. We should tackle that when it comes.

Reply:
I had the same thought, but to make progress with this was happy that the narrowly scoped interface was sufficient to minimise the coupling. I definitely didn't want to bloat the application state.

Reply:
I think this is fine. Once we remove RunAtomic, this might become a dependency of application state (rather than application service).

I'd be OK with states with the same database having dependency on a subset of functionality of another state. As long as those are true dependencies and are acyclic.

Sometime soon we will be faced with redundant complex code, we should deal with that either through DI, common packages or code gen.